### ISSUE TITLE
not triggering `no-extraneous-require` for submodules

### ISSUE THREAD

**Issue by alex-Symbroson** [mysticatea/eslint-plugin-node#335]
I have a project where I am requiring individual files of a self created module.

Example:
Module has following structure:
```
mymodule/
- package.json
- index.js
- part.js
```
and my other project only requires part.js:
```js
const part = require("mymodule/part");
```

What would be the correct steps to not trigger the no-extraneous-require rule?
Or is this considered a bug that the rule doesn't specifically check for this?

**Comment by alex-Symbroson** [comment]
false report
